build.zig

```diff
@@ -48,6 +48,21 @@ pub fn build(b: *zbs.Builder) !void {
 
     const examples = b.option(bool, "examples", "Set to true to build examples") orelse false;
 
+    const version = v: {
+        const version_string = "0.1.0";
+
+        var ret: u8 = undefined;
+        const git_dir = fs.path.join(b.allocator, &[_][]const u8{ b.build_root, "/.git" }) catch unreachable;
+        const git_commit_hash = b.execAllowFail(
+            &[_][]const u8{ "git", "--git-dir", git_dir, "--work-tree", b.build_root, "rev-parse", "--short", "HEAD" },
+            &ret,
+            .Ignore,
+        ) catch break :v version_string;
+
+        const dev_version = b.fmt("{s}-dev-{s}", .{ version_string, git_commit_hash });
```

Comment:
let's use `std.fmt.allocPrintZ()` instead here and get rid of the `dupeZ()` calls below.